### ISSUE TITLE
SentrySpan does not have a `data` field

### ISSUE THREAD

**Issue by marandaneto** [getsentry/sentry-java#1538]
https://develop.sentry.dev/sdk/event-payloads/span/

```
{
      "data": {
        "type": "xhr",
        "method": "GET",
        "url": "https://api.covid19api.com/summary"
      },
      "description": "GET https://api.covid19api.com/summary",
      "op": "http",
      "parent_span_id": "ae1192f4fbda20ff",
      "span_id": "9f27dcdf0a77661e",
      "start_timestamp": 1623849813.575222,
      "status": "ok",
      "tags": {
        "http.status_code": "200"
      },
      "timestamp": 1623849814.675915,
      "trace_id": "1e5acd8e53254c9db2be50cb40e9ada4"
    }
```

during ser/deser, we're losing the `data` field

**Comment by maciejwalkowiak** [comment]
There's just no `data` field on `SentrySpan`. What should be the type? `Map<String, String>` or `Map<String, Object>`? Does the backend handle any JSON passed in `data`?

**Comment by marandaneto** [comment]
looks like similar to https://github.com/getsentry/sentry-java/issues/1233
maybe we don't expose a public API to add such data, but lets be sure that it survives a round trip ser/deser.
based on the sample, looks like `Map<String, Object>`, because there are String and Int:

```
{
  "data": {
    "url": "http://localhost:8080/sockjs-node/info?t=1588601703755",
    "status_code": 200,
    "type": "xhr",
    "method": "GET"
  }
}
```

**Comment by maciejwalkowiak** [comment]
Can be done, but if we want it to survive serialization, why not adding it to public API?

**Comment by marandaneto** [comment]
> Can be done, but if we want it to survive serialization, why not adding it to public API?

the reason is the same as before, https://github.com/getsentry/sentry-java/issues/1233 if we expose `setData(x)` to `ISpan`, it leaks to `ITransaction` but `ITransaction` does not have a `data` field.
this should be solved with composition over inheritance probably, but it'd require a bigger refactoring and breaking changes, most probably.

**Comment by bruno-garcia** [comment]
Yeah and given we're rethinking this API now its' better we avoid expanding the public API (even though Data could just use Extra as I believe is done on other SDKs).

We plan to change to change serialization (see #1537 ) that will address such cases where we don't map completely to all other Sentry SDKs but for the time being to address this React Native issue would be nice to just special case `Data` as @marandaneto suggested

**Comment by marandaneto** [comment]
let's do a quick fix then before we get into #1537 which is gonna take longer, so we don't lose the `data` field during round trip ser/deser
